presto-oracle/src/main/java/io/prestosql/plugin/oracle/OracleClient.java

```diff
@@ -0,0 +1,125 @@
+            case Types.SMALLINT:
+                return Optional.of(StandardColumnMappings.smallintColumnMapping());
+            case Types.INTEGER:
+                return Optional.of(StandardColumnMappings.integerColumnMapping());
+            case Types.BIGINT:
+                return Optional.of(StandardColumnMappings.bigintColumnMapping());
+            case Types.REAL:
+                return Optional.of(StandardColumnMappings.realColumnMapping());
+            case Types.FLOAT:
+            case Types.DOUBLE:
+                return Optional.of(StandardColumnMappings.doubleColumnMapping());
+            case Types.BLOB:
+            case Types.BINARY:
+            case Types.VARBINARY:
+            case Types.LONGVARBINARY:
+                return Optional.of(StandardColumnMappings.varbinaryColumnMapping());
+            case Types.DATE:
+                return Optional.of(StandardColumnMappings.dateColumnMapping());
+            case Types.TIME:
+                return Optional.of(StandardColumnMappings.timeColumnMapping());
+            case Types.TIMESTAMP:
+                return Optional.of(StandardColumnMappings.timestampColumnMapping(session));
+            case Types.NUMERIC:
+                return Optional.of(StandardColumnMappings.decimalColumnMapping(createDecimalType(columnSize == 0 ? 38 : columnSize, max(typeHandle.getDecimalDigits(), 0))));
+            case Types.DECIMAL:
+                int decimalDigits = typeHandle.getDecimalDigits();
+                int precision = columnSize + max(-decimalDigits, 0); // Map decimal(p, -s) (negative scale) to decimal(p+s, 0).
+                if (precision > Decimals.MAX_PRECISION) {
+                    return Optional.empty();
+                }
+                return Optional.of(StandardColumnMappings.decimalColumnMapping(createDecimalType(precision, max(decimalDigits, 0))));
+            case Types.CHAR:
+            case Types.NCHAR:
+                // TODO this is wrong, we're going to construct malformed Slice representation if source > charLength
+                int charLength = min(columnSize, CharType.MAX_LENGTH);
+                return Optional.of(StandardColumnMappings.charColumnMapping(createCharType(charLength)));
+            case Types.VARCHAR:
+            case Types.NVARCHAR:
+            case Types.LONGVARCHAR:
+            case Types.LONGNVARCHAR:
+                return (columnSize > VarcharType.MAX_LENGTH || columnSize == 0)
+                    ? Optional.of(StandardColumnMappings.varcharColumnMapping(createUnboundedVarcharType()))
+                    : Optional.of(StandardColumnMappings.varcharColumnMapping(createVarcharType(columnSize)));
+            default:
+                Optional<ColumnMapping> mapping = this.getForcedMappingToVarchar(typeHandle);
+                return mapping.isPresent() ? mapping : StandardColumnMappings.jdbcTypeToPrestoType(session, typeHandle);
```

Comment:
We don't need to write all type mapping here. You can see the existing implementation in PostgreSqlClient.

Reply:
done

Reply:
Most types are still unnecessary in the case condition because `super.toPrestoType` contains it. Please write only Oracle specific condition here.

Reply:
fixed now, sorry